Sources/Paywalls/PaywallData.swift

```diff
@@ -15,6 +15,37 @@
 
 import Foundation
 
+/// Represents countries where currencies typically have zero decimal places
+public struct ZeroDecimalPlaceCountries: Codable, Sendable, Hashable, Equatable {
+
+    /// Storefront IDs that should typically display zero decimal places
+    public var apple: Set<String> {
```

Comment:
Probably a bit late to this, but a couple questions:
- This list of countries would be static right? And it probably depends on the currency, not the country, I would think? Just wondering if this info would be something that is ok to hardcode in the SDK...
- In case the above is wrong, I believe we send the storefront as a header in all requests? We could probably have the backend tell us whether to do this, instead of sending a whole list of countries... But we can continue with this for now if the backend has already been implemented.

Reply:
> This list of countries would be static right? And it probably depends on the currency, not the country, I would think? Just wondering if this info would be something that is ok to hardcode in the SDK.

Well doing this on the currency _is_ correct, the storefront also determines the currency 🤷‍♂️ I guess `StoreProduct` does have `currencyCode` but it like there could be some discrepancy between what the underlying SK1 and SK2 return there while storefront is pretty well documented as `ISO 3166-1 Alpha-3`

Reply:
I think country is better than currency here, because it could be customary to write prices as "$7" in some countries that use dollars while maybe "$7.00" could be customary in others. Using country gives more granular control than using currency. But what could be missing here is the ability to round for non-computed values **only** if it's close to a whole number. But I suppose that is another issue....?

Reply:
Well, I think it should be ok to use countries. Wdyt about my other suggestions? That is, hardcoding the countries in the SDK instead of getting them from the backend?

Reply:
For currencies lacking minor units, wouldn't "7.00" be incorrect in any country using that currency? We'd be inventing "cents" which don't exist in that currency. 

I'd be inclined to vote for using currencies too because of this reason, but feel free to ignore me, especially if it's too late to change now.

Reply:
> That is, hardcoding the countries in the SDK instead of getting them from the backend

Trying to avoid hardcoding as many things in the SDK for paywalls as possible 😅 Activity trying to pull out the localizations we have hardcoded (for restore, terms, privacy, durations, etc) because there have been so many requested updates and its hard to say which version things are/aren't fixed/supported in (especially across multiple SDKs).

And the API trip is already being made so its mostly free to do this and I'd prefer to have more control if we need to add or remove any 🤷‍♂️ 

> I'd be inclined to vote for using currencies too because of this reason, but feel free to ignore me, especially if it's too late to change now.

FWIW, our backend also has a list of countries with zero decimal places by country 🤷‍♂️  I'm not an super expert on currencies/country formatting though

Reply:
>Trying to avoid hardcoding as many things in the SDK for paywalls as possible
 
Currencies, time zone offsets, these things are [not](https://en.wikipedia.org/wiki/Croatia_and_the_euro#:~:text=Croatia%20adopted%20the%20euro%20as,member%20state%20of%20the%20eurozone.) [set](https://commission.europa.eu/news/commission-assesses-countries-progress-towards-adopting-euro-2024-06-26_en) in [stone](https://en.wikipedia.org/wiki/Bitcoin_in_El_Salvador). So would indeed be good to be able to update from the backend. 
  
Regarding currency/country formatting, my understanding is that lacking minor units is an attribute of the currency. Formatting rules on the other hand are country (locale) dependent. E.g. 7 US dollars is `$7.00` in `enUS` and `US$ 7,00` in `nlNL`. So whether a currency has minor units (decimals) and how to format a currency are different (but related) things. Again, I don't want to be the person needlessly dragging this on. So if the current approach works, it works 😄

Reply:
My $.02:
I don’t have a strong feeling either way. 

**Storefront vs currency:**
My intuition is that it’d be more tied to currency than country, but I haven’t done the research. 
Storefront is more granular, though, right? Because a currency can be in multiple storefronts but not the other way around, so using storefront seems more flexible. So the cost when going with storefront instead of currency is a _little_ bit more duplicated information but the gain is extra flexibility for us to fix things if we've made a wrong call. 

**Backend-driven vs hardcoded:**
Backend-driven seems like a good idea, since it makes things way easier to fix if we make wrong assumptions, which as we learned from localization stuff, we tend to do. It's been really painful for us to fix our bad localizations in the SDK, and even more painful for customers to deal with them and to have to update the SDK to get fixes.

So I'd imagine that storefront + backend-driven is the winning combination.

Reply:
This actually started because in Taiwan, their currency does have cents, but those are typically only seen written in prices at gas stations and dumpling shops. Including the cent component on the paywalls looked weird.

You could make the argument that all locales using the same currency have the same customs when it comes to including cents in written prices or not, but I don't know if that's true, so as Andy wrote above, storefronts is the flexible/safe bet.

Reply:
Ah nice, TIL, thanks! It's always the exceptions 😅